src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -717,6 +725,7 @@ protected virtual void Dispose(bool disposing)
                 }
             }
 
+            _componentStateById.Clear(); // So we know they were all disposed
```

Comment:
We already had logic that no-ops a render request if the component was already disposed. Adding this line means that when the entire renderer itself is disposed, we treat all its components as being disposed (which they are) so the existing no-op logic takes care of it.

Reply:
Am I correct in understanding that Dispose always run within the sync context?

Reply:
`CircuitHost` does call `Renderer.Dispose` from within an `Invoke` callback, so yes I think so. I'm not aware of other places where server-side Blazor disposes the renderer.